### PR TITLE
chore: simplify type-coverage configuration

### PR DESCRIPTION
**Pull request by septs**
https://github.com/plantain-00/type-coverage/issues/88

### CODE REVIEW

**Review by guanbinrui — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -70,17 +70,30 @@
...
   },
-  "engines": {
-    "node": ">=14.4.0",
-    "pnpm": ">=5"
+  "typeCoverage": {
+    "cache": true,
+    "strict": true,
+    "detail": true,
+    "atLeast": 98,
+    "project": "packages/maskbook",
```
Comment (Jack-Works) [inline package.json:91]: should we lint the whole repo?

File: package.json [hunk 2]
```diff
@@ -70,17 +70,30 @@
     "prettier": "^2.1.2",
     "ts-node": "^9.1.1",
     "ttypescript": "^1.5.12",
-    "type-coverage": "^2.14.8",
+    "type-coverage": "^2.17.0",
     "typescript": "4.1.3"
   },
+  "engines": {
+    "node": ">=14.4.0",
+    "pnpm": ">=5"
```
Comment (Jack-Works) [inline package.json:78]: I think we should add npm: >=999 and yarn: >=999 to prevent people from using those tools

File: package.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
     "build:ios:webpack": "build iOS",
     "lint": "eslint -c ./packages/.eslintrc.json packages/ --ext .ts,.tsx --fix --cache",
     "lint:report": "eslint -c ./packages/.eslintrc.json packages/ --ext .ts,.tsx --format junit -o reports/junit/eslint-results.xml --cache",
-    "lint:typecoverage": "type-coverage --strict -p packages/maskbook --cache --ignore-catch --ignore-files packages/maskbook/src/plugins/Wallet/contracts/** --ignore-files **/__tests__/** --ignore-files **\\__tests__\\** --detail --at-least 98",
```
Comment (septs) [inline package.json:21]: @Jack-Works origin command project lint on `packages/maskbook`

```
-p packages/maskbook
```

File: package.json (hunk 2 above)
Comment (septs) [reply package.json:78]: new pr handle this problem?

File: package.json (hunk 1 above)
Comment (septs) [reply package.json:91]: new pr handle this problem?

**Review by Jack-Works — APPROVED** [review]

**Pull request merged**